pkg/render/common/networkpolicy/networkpolicy.go

```diff
@@ -92,6 +93,27 @@ func CreateSourceEntityRule(namespace string, deploymentName string) v3.EntityRu
 	}
 }
 
+// GetOIDCEgressRule creates egress rule for oidc connection.
+// the result will include an egress rules with the urlString passed in:
+//  1. egress rule: egress rule assuming the oidc is external to the cluster
+func GetOIDCEgressRule(urlString string) v3.Rule {
+	parsedURL, err := url.Parse(urlString)
```

Comment:
Do we check already somewhere that this can be parsed? We don't want this to ever happen.

Reply:
I think this has to be check when initializing the authentication cr values. Let me check it.